lib/cpp/src/thrift/protocol/TProtocol.h

```diff
@@ -659,6 +659,11 @@ class TProtocolFactory {
   virtual ~TProtocolFactory() {}
 
   virtual boost::shared_ptr<TProtocol> getProtocol(boost::shared_ptr<TTransport> trans) = 0;
+  virtual boost::shared_ptr<TProtocol> getProtocol(boost::shared_ptr<TTransport> inTrans,
```

Comment:
I'm not sure why this getProtocol overload is here.  All the servers have an inputProtocolFactory and outputProtocolFactory.  Maybe this is a little bit more usable... but it adds unusual noise to the review.  I would like to have some justification for this overload in regards to THeader... or I would like to see this pulled into its own issue.